scapy/contrib/automotive/uds.py

```diff
@@ -1221,6 +1224,103 @@ def get_log(pkt):
 bind_layers(UDS, UDS_RTEPR, service=0x77)
 
 
+# #########################RFT###################################
+class UDS_RFT(Packet):
+    name = 'RequestFileTransfer'
+
+    modeOfOperations = {
+        0x00: "ISO/SAE Reserved",
+        0x01: "Add File",
+        0x02: "Delete File",
+        0x03: "Replace File",
+        0x04: "Read File",
+        0x05: "Read Directory"
+    }
+
+    @staticmethod
+    def _contains_file_size(packet):
+        return packet.modeOfOperation not in [2, 4, 5]
+
+    fields_desc = [
+        XByteEnumField('modeOfOperation', 0, modeOfOperations),
+        FieldLenField('filePathAndNameLength', 0,
+                      length_of='filePathAndName', fmt='H'),
+        StrLenField('filePathAndName', b"",
+                    length_from=lambda p: p.filePathAndNameLength),
+        ConditionalField(BitField('compressionMethod', 0, 4),
+                         lambda p: p.modeOfOperation not in [2, 5]),
+        ConditionalField(BitField('encryptingMethod', 0, 4),
+                         lambda p: p.modeOfOperation not in [2, 5]),
+        # TODO: Check if FieldLenField can be assigned for two fields
```

Comment:
@guedou 
The lengths `fileSizeUnCompressed` and `fileSizeCompressed` depend on `fileSizeParameterLength`.

But it seems that I can give only one element to `length_of` of `FieldLenField`.
Is there a common workaround with mimics something like
```python
length_of=['fileSizeUnCompressed', 'fileSizeCompressed']
```

Reply:
I would just apply it to the first field, as it is, right now. From  what I see in the code, length_of is used to determine the field value of this field. Since both fields should have the same size it would not matter from which one the length is taken.